| 20231111
|       + modify endwin() to return an error if it is called again without an
|         intervening screen update (report by Rajeev Pillai, NetBSD #57592).
--- a/ranger/gui/ui.py
+++ b/ranger/gui/ui.py
@@ -147,6 +147,7 @@ class UI(  # pylint: disable=too-many-in
             pass
         if self.settings.mouse_enabled:
             _setup_mouse(dict(value=False))
+        self.win.refresh()
         curses.endwin()
         self.is_on = False
 
